frontend/src/components/NavBar.tsx

```diff
@@ -31,26 +31,29 @@ export const NavBar = () => {
         ) : (
           <LoginNavBar type="register" />
         ))}
-      {userInfo && (
-        <div className={classes(css.navBarDiv)}>
-          <CornerPiece />
-          <div className={classes(css.navBarRightSide)}>
-            <>
-              <div className={classes(css.navBarText)}>
-                <Trans i18nKey="Project" />
-              </div>
-              <RoadmapSelectorWidget />
-              <div className={classes(css.navBarDivider)} />
-              <Link className={classes(css.navBarLink)} to={paths.userInfo}>
-                <PermIdentityIcon className={classes(css.icon)} />
-              </Link>
-              <Link className={classes(css.navBarLink)} to={paths.logoutPage}>
-                <PowerSettingsNewIcon className={classes(css.icon)} />
-              </Link>
-            </>
+      {userInfo &&
```

Comment:
I think this whole component could be refactored a bit

Using couple conditional early returns would make this much clearer in my opinion, e.g.

```ts
  if (!userInfo) {
    return pathname.startsWith(paths.loginPage) ? (
      <LoginNavBar type="login" />
    ) : (
      <LoginNavBar type="register" />
    );
  }

  if (pathname.startsWith(paths.getStarted))
    return <LoginNavBar type="getStarted" />;

  return (
    <div className={classes(css.navBarDiv)}>
      <CornerPiece />
      <div className={classes(css.navBarRightSide)}>
        <div className={classes(css.navBarText)}>
          <Trans i18nKey="Project" />
        </div>
        <RoadmapSelectorWidget />
        <div className={classes(css.navBarDivider)} />
        <Link className={classes(css.navBarLink)} to={paths.userInfo}>
          <PermIdentityIcon className={classes(css.icon)} />
        </Link>
        <Link className={classes(css.navBarLink)} to={paths.logoutPage}>
          <PowerSettingsNewIcon className={classes(css.icon)} />
        </Link>
      </div>
    </div>
  );
```